src/test_workflow/test_cluster.py

```diff
@@ -87,6 +87,7 @@ def terminate(self) -> None:
         for service in self.dependencies:
             termination_result = service.terminate()
             self.__save_test_result_data(termination_result)
+            service.uninstall()
```

Comment:
Could you make sure the uninstallation of dependencies wont affect `__save_test_result_data`?